src/session.cpp

```diff
@@ -287,6 +287,45 @@ namespace libtorrent
 	// configurations this will give a link error
 	void TORRENT_CFG() {}
 
+	session_params read_session_params(bdecode_node const& e, std::uint32_t const flags)
```

Comment:
I wonder if it would make sense for this function to take an ``error_code&``. I believe ``read_resume_data()``, does, and I feel like this is the same category of operation.